src/App.tsx

```diff
@@ -1,61 +1,103 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [option, setOption] = useState(0);
+  const [title, setTitle] = useState('');
+  const [initialTodos, setNewTodo] = useState(todosFromServer
+    .map(todo => ({ ...todo, user: getUser(todo.userId) })));
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newTodo: Todo = {
+      id: Math.max(...initialTodos.map(todo => todo.id)) + 1,
+      title,
+      completed: false,
+      userId: option,
+      user: getUser(option),
+    };
+
```

Comment:
add validation for inputs
![Screenshot from 2023-04-01 17-25-43](https://user-images.githubusercontent.com/33029864/229295399-20dc51f4-7276-4c93-ae15-234c0c53f059.png)